### PR TITLE
Feature/set up redux store

### PR DESCRIPTION
**Pull request by emarkees**
In this project
- React-Redux was setUp
-  Redux store was setUp
- States were declared
- Application files were structured following the use of [ducks pattern](https://github.com/erikras/ducks-modular-redux)

Kindly help me review

Thanks

### CODE REVIEW

**Review by KanzaTahreem — COMMENTED** [review]
# INVALID ❌ 

Hi @emarkees  👋

While you made a great effort in this project, unfortunately, I cannot proceed to review your code. 😓

## Invalid Code Review Request ❌
- [ ] Unfortunately, you have submitted a pull request that lacks a Microverse [README template](https://github.com/microverseinc/readme-template/blob/master/README.md). Please include it and don't forget to customize the README content according to the project you are working on 🤓⛏

_**Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready**_ 🤝🙏

Cheers and Happy coding!👏👨‍💻💻

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+const initialState = {
+  categories: [],
+  status: 'Under construction',
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {},
+});
```
Comment (AmaduKamara) [inline src/redux/categories/categoriesSlice.js]: - [x] It's required that you create one reducer for checking the status of categories. Kindly add that.
```js
reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
```

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export const { action: categoriesActions, reducer: categoriesReducer } = categoriesSlice;
```
Comment (AmaduKamara) [inline src/redux/categories/categoriesSlice.js]: - [x] Kindly export your reducer and action as required. You are exporting action and reducer that are not present in this file. After adding the reducer above, kindly export the reducer and action like this below:

```javascript
export const { checkStatus } = categoriesSlice.actions

export default categoriesSlice.reducer
```

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @emarkees,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽 
✅ Used Redux Toolkit as required 👍🏽

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/redux/categories/categoriesSlice.js (hunk 2 above)
Comment (emarkees) [reply src/redux/categories/categoriesSlice.js]: Correction done

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (emarkees) [reply src/redux/categories/categoriesSlice.js]: done

**Review by roshan-bajgain — APPROVED** [review]
Hi @emarkees, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Good job on implementing the changes ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 2 above)
Comment (emarkees) [reply src/redux/categories/categoriesSlice.js]: Thank you @AmaduKamara

**Comment by emarkees** [comment]
Thanks you @roshan-bajgain

**Pull request merged**